internal/blockchain/ethereum/quorum/quorum.go

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2024 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quorum
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hyperledger/firefly-cli/internal/docker"
+)
+
+func CreateQuorumEntrypoint(ctx context.Context, outputDirectory, volumeName, memberIndex, consensus, stackName string, chainId int, tesseraEnabled bool) error {
+	discoveryCmd := "BOOTNODE_CMD=\"\""
+	connectTimeout := 15
+	if memberIndex != "0" {
+		discoveryCmd = fmt.Sprintf(`bootnode=$(curl http://geth_0:%s -s --connect-timeout %[2]d --max-time %[2]d --retry 5 --retry-connrefused --retry-delay 0 --retry-max-time 60 --fail --header "Content-Type: application/json" --data '{"jsonrpc":"2.0", "method": "admin_nodeInfo", "params": [], "id": 1}' | grep -o "enode://[a-z0-9@.:]*")
+BOOTNODE_CMD="--bootnodes $bootnode"
+BOOTNODE_CMD=${BOOTNODE_CMD/127.0.0.1/geth_0}`, GethPort, connectTimeout)
+	}
+
+	tesseraCmd := ""
+	if tesseraEnabled {
+		tesseraCmd = fmt.Sprintf(`TESSERA_URL=http://%[5]s_member%[1]stessera
+TESSERA_TP_PORT=%[2]s
+TESSERA_Q2T_PORT=%[3]s
+TESSERA_UPCHECK_URL=$TESSERA_URL:$TESSERA_TP_PORT/upcheck
+ADDITIONAL_ARGS="${ADDITIONAL_ARGS:-} --ptm.timeout 5 --ptm.url ${TESSERA_URL}:${TESSERA_Q2T_PORT} --ptm.http.writebuffersize 4096 --ptm.http.readbuffersize 4096 --ptm.tls.mode off"
+
+echo -n "Checking tessera is up ... "
+curl --connect-timeout %[4]d --max-time %[4]d --retry 5 --retry-connrefused --retry-delay 0 --retry-max-time 60 --silent --fail "${TESSERA_UPCHECK_URL}"
+echo ""
+`, memberIndex, TmTpPort, TmQ2tPort, connectTimeout, stackName)
+	}
+
+	content := fmt.Sprintf(`#!/bin/sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+GOQUORUM_CONS_ALGO=%s
+if [ "istanbul" == "$GOQUORUM_CONS_ALGO" ];
+then
+    echo "Using istanbul for consensus algorithm..."
+    export CONSENSUS_ARGS="--istanbul.blockperiod 5 --mine --miner.threads 1 --miner.gasprice 0 --emitcheckpoints"
```

Comment:
uhmm this block period is not using the configuration... I assume this overwrites the one in the genesis

Reply:
my bad, going to pass it in as a parameter into the function. Clique and QBFT takes from Genesis file, other options allows CLI flags